pkg/gds/admin.go

```diff
@@ -239,6 +242,68 @@ func (s *Admin) ListVASPs(c *gin.Context) {
 	c.JSON(http.StatusOK, out)
 }
 
+func (s *Admin) RetrieveVASP(c *gin.Context) {
+	var (
+		err    error
+		vaspID string
+		vasp   *pb.VASP
+		out    *admin.RetrieveVASPReply
+	)
+	// Get vaspID from the URL
+	vaspID = c.Param("vaspID")
+
+	// Attempt to fetch the VASP from the database
+	if vasp, err = s.db.RetrieveVASP(vaspID); err != nil {
+		log.Warn().Err(err).Str("id", vaspID).Msg("could not retrieve vasp")
+		c.JSON(http.StatusNotFound, admin.ErrorResponse("could not retrieve VASP record by ID"))
+		return
+	}
+
+	// Create the response to send back
+	out = &admin.RetrieveVASPReply{
+		VerifiedContacts: models.VerifiedContacts(vasp),
+		Traveler:         models.IsTraveler(vasp),
+	}
+	if out.Name, err = vasp.Name(); err != nil {
+		log.Warn().Err(err).Msg("could not get VASP name")
```

Comment:
Could we bump this up to an error? If a record got into GDS and doesn't have any name at all, that seems bad